common/network/interface.go

```diff
@@ -0,0 +1,29 @@
+package network
+
+import (
+	"fmt"
+	"net"
+)
+
+// GetFirstAddressOf returns the first address of the supplied interface name.
+func GetFirstAddressOf(name string) (string, error) {
```

Comment:
Is this always going to return the right (same) address? i.e. might it sometimes return the ipv6 addr?